### PR TITLE
Add custom validation

### PR DESCRIPTION
**Pull request by kul3r4**
Solves #598

### CODE REVIEW

File: src/20_Components/amp-form.html [hunk 1]
```diff
@@ -70,8 +70,16 @@
     The `amp-form` component displays `submit-success` or `submit-error` elements based on the response and
     renders the template data inside these two elements. The `submit-success` and `submit-error` elements
     must be direct children of `form`.
+
```
Comment (sebastianbenz) [inline src/20_Components/amp-form.html]: We should add a heading before to make it appear in the outline

**Comment by kul3r4** [comment]
Sure will do!

**Comment by kul3r4** [comment]
@sebastianbenz PTAL

**Comment by kul3r4** [comment]
There is also [amp-form: async verification](https://github.com/ampproject/amphtml/issues/9700) that could be added, @aghassemi was interested in that, we could do it in a separate PR.

File: src/20_Components/amp-form.html [hunk 2]
```diff
@@ -71,10 +71,126 @@
...
+      <template type="amp-mustache">
+        Error! Thanks {{name}} for trying the <code>amp-form</code> demo with an error response.
+      </template>
+    </div>
+  </form>
+
+  <!-- ## Form Custom Validation -->
+  <!--
+    The `amp-form` extension allows you to build your own custom validation UI by using the `custom-validation-reporting`,
+  read [here](https://www.ampproject.org/docs/reference/components/amp-form#custom-validations) about the possible strategies.
```
Comment (sebastianbenz) [inline src/20_Components/amp-form.html]: s/possible/different

**Pull request merged**